### PR TITLE
unitTests: Add tests for ocs to ocs provider side code

### PR DESCRIPTION
**Pull request by iamniting**
Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

File: controllers/storagecluster/provider_server_test.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+		assert.Error(t, err)
+		assert.Equal(t, "Deployment ocs-provider-server is not ready", err.Error())
+
+		deployment := &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{Name: ocsProviderServerName},
+		}
+		assertFound(t, r.Client, deployment)
+		expectedDeployment := GetProviderAPIServerDeployment(instance)
+		assert.Equal(t, deployment.Spec, expectedDeployment.Spec)
```
Comment (nb-ohad) [inline controllers/storagecluster/provider_server_test.go:38]: Shouldn't this need to be deep equal? or is `assert.equal` has deep equal semantics?

File: controllers/storagecluster/provider_server_test.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+		service := &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{Name: ocsProviderServerName},
+		}
+		assertFound(t, r.Client, service)
+		expectedService := GetProviderAPIServerService(instance)
+		assert.Equal(t, expectedService.Spec, service.Spec)
```
Comment (nb-ohad) [inline controllers/storagecluster/provider_server_test.go:45]: same as last comment

File: controllers/storagecluster/provider_server_test.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+		ObjectMeta: metav1.ObjectMeta{Name: ocsProviderServerName},
+	}
+	assertNotFound(t, cli, service)
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{Name: ocsProviderServerName},
+	}
+	assertNotFound(t, cli, secret)
+
+}
```
Comment (nb-ohad) [inline controllers/storagecluster/provider_server_test.go:131]: This function is used by a single test and is very specific, can we just move it into the test itself? 
Right now the reader needs to jump between two distinctive locations in the source code for no real reason.

File: controllers/storagecluster/provider_server_test.go [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+	})
+}
+
+func assertFound(t *testing.T, cli client.Client, object client.Object) {
+
+	key := client.ObjectKeyFromObject(object)
+
+	err := cli.Get(context.TODO(), key, object)
+	assert.NoError(t, err)
+}
```
Comment (nb-ohad) [inline controllers/storagecluster/provider_server_test.go:103]: Why do we need this util when we can just do this (in the call site) instead?
```suggestion
assert.NoError(t, cli.Get(context.TODO(), client.ObjectKeyFromObject(object), object))
```

File: controllers/storagecluster/provider_server_test.go [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+func assertNotFound(t *testing.T, cli client.Client, object client.Object) {
+
+	key := client.ObjectKeyFromObject(object)
+
+	err := cli.Get(context.TODO(), key, object)
+	assert.Error(t, err)
+	assert.True(t, errors.IsNotFound(err))
+}
```
Comment (nb-ohad) [inline controllers/storagecluster/provider_server_test.go:112]: `assert.True(t, errors.IsNotFound(err))` is already testing for `assert.Error(t, err)` and the entire thing can be compressed into 
```suggestion
assert.True(t, errors.IsNotFound(cli.Get(context.TODO(), client.ObjectKeyFromObject(object), object)))
```

File: controllers/storagecluster/provider_server_test.go (hunk 1 above)
Comment (iamniting) [reply controllers/storagecluster/provider_server_test.go:38]: It does verify it.

File: controllers/storagecluster/provider_server_test.go (hunk 2 above)
Comment (iamniting) [reply controllers/storagecluster/provider_server_test.go:45]: same as prev

File: controllers/storagecluster/provider_server_test.go [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+		_, err := obj.ensureCreated(r, instance)
+
+		assert.Error(t, err)
+		assert.Equal(t, "Deployment ocs-provider-server is not ready", err.Error())
+
+		deployment := &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{Name: ocsProviderServerName},
+		}
+		assertFound(t, r.Client, deployment)
+		expectedDeployment := GetProviderAPIServerDeployment(instance)
```
Comment (nb-ohad) [inline controllers/storagecluster/provider_server_test.go:37]: Why are we testing against data coming from the code base itself? what if we have a bug in `GetProviderAPIServerDeployment`?

**Review by nb-ohad — APPROVED** [review]

**Pull request merged**